sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ColumnResolutionHelper.scala

```diff
@@ -487,42 +487,81 @@ trait ColumnResolutionHelper extends Logging with DataTypeErrorsBase {
   //       original expression as it is.
   private def tryResolveColumnByPlanId(
```

Comment:
let's re-think about the naming. I think this one can be `tryResolveDataFrameColumns`, then it calls `resolveDataFrameColumn`, which then calls `resolveDataFrameColumnByPlanId`, which then calls `resolveDataFrameColumnRecursively`